scripts/monorepo/runPublished.spec.ts

```diff
@@ -0,0 +1,124 @@
+import childProcess from 'child_process';
+
+import main from './runPublished';
+
+// ☝️ NOTE - don't comment or remove this mock while running tests - you will need to manually kill lot of node processes :P
+jest.mock('child_process');
+
+describe(`#runPublished`, () => {
+  const originalConsoleLog = console.log;
+  beforeEach(() => {
+    console.log = () => {};
+  });
+  afterAll(() => {
+    console.log = originalConsoleLog;
+  });
+
+  function setup() {
+    const workspacePackagesMetadata = {
+      '@fluentui/web-components': {
+        packagePath: 'packages/web-components',
+        packageJson: { name: '@fluentui/web-components', version: '1.0.0', main: 'lib/index.js', private: false },
+      },
+      '@fluentui/react': {
+        packagePath: 'packages/react',
+        packageJson: { name: '@fluentui/react', version: '8.0.0', main: 'lib/index.js', private: false },
+      },
+      '@fluentui/react-northstar': {
+        packagePath: 'packages/fluentui/react-northstar',
+        packageJson: { name: '@fluentui/react-northstar', version: '0.1.0', main: 'lib/index.js', private: false },
+      },
+      '@fluentui/react-components': {
+        packagePath: 'packages/react-components/react-components',
+        packageJson: { name: '@fluentui/react-components', version: '9.0.0', main: 'lib/index.js', private: false },
+      },
+    };
+
+    const spawnSyncMock = jest.spyOn(childProcess, 'spawnSync').mockImplementation((() => {
+      return { status: 0 };
+    }) as any);
+
+    jest.spyOn(process, 'exit').mockImplementation((() => {}) as any);
+
+    function getSpawnCallArguments() {
+      const [, spawnSyncArgs, spawnSyncOptions] = spawnSyncMock.mock.calls[0];
+      const [lagePath, ...restArgs] = spawnSyncArgs as string[];
+
+      return { spawnSyncArgs, spawnSyncOptions, lagePath, restArgs };
+    }
+
+    return { workspacePackagesMetadata, spawnSyncMock, getSpawnCallArguments };
+  }
+
+  it(`should not do invoke lage with '--to' if all packages are private`, () => {
+    const { workspacePackagesMetadata, getSpawnCallArguments } = setup();
+    const metadataCopy = { ...workspacePackagesMetadata };
+    metadataCopy['@fluentui/react'].packageJson.private = true;
+    metadataCopy['@fluentui/react-components'].packageJson.private = true;
+
+    main({
+      argv: ['build'],
+      workspacePackagesMetadata: metadataCopy,
+    });
+
+    const { restArgs } = getSpawnCallArguments();
+
+    expect(restArgs).toMatchInlineSnapshot(`
```

Comment:
this feels like a bug, IMO nothing should be run if there are no packages "affected" cc @ling1726